packages/ui/src/utils/VisibilityObserver.tsx

```diff
@@ -40,8 +40,22 @@ export function useVisibilityObserver<T extends Element>(
 
     observer.observe(node);
 
+    function handleVisibilityChange(): void {
```

Comment:
How about adding this check [here](https://github.com/superdispatch/web-ui/pull/513/files#diff-18c3bd689875870afb3534111c53dec6e76c1f419581419dd8e10736fc5fa789R36)?

Reply:
> setState(entry?.isIntersecting ? 'visible' : 'invisible');

Reply:
anyway, I like your suggestion, it makes less code change.